### PR TITLE
[CPU] If operation implementation.

### PR DESCRIPTION
**Pull request by nshchego**
### Details:
 - *'If' operation enabling.*

### Tickets:
 - *50052*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@mandrono Please review the PR

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine::details;
+
+namespace MKLDNNPlugin {
+
+static NodeConfig makePlainConfig(const std::shared_ptr<ngraph::Node>& op) {
+    NodeConfig config;
+
+    for (size_t i = 0; i < op->get_input_size(); i++) {
+        const auto& dims = op->get_input_shape(i);
```
[the remark this answers is not in the document]
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp]: Done

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    }
+    const std::shared_ptr<const ngraph::Function>& thenBody = ifOp->get_then_body();
+    const std::shared_ptr<const ngraph::Function>& elseBody = ifOp->get_else_body();
+    subGraphThen.CreateGraph(thenBody, ext_mng, weightCache);
+    subGraphElse.CreateGraph(elseBody, ext_mng, weightCache);
+
+    const auto &inMapThen = subGraphThen.GetInputNodesMap();
+    for (const auto &param : ifOp->get_then_body()->get_parameters()) {
+        auto inNode = inMapThen.find(param->get_friendly_name());
+        if (inNode != inMapThen.end()) {
```
[the remark this answers is not in the document]
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp:96]: Done

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    void initSupportedPrimitiveDescriptors() override;
+    void getSupportedDescriptors() override;
+    void createPrimitive() override;
+    bool created() const override;
+    void execute(mkldnn::stream strm) override;
+
+    void inline setExtManager(const MKLDNNExtensionManager::Ptr& extMgr) { ext_mng = extMgr; }
+
+    struct PortMap {
+        int from;
```
[the remark this answers is not in the document]
Comment (nshchego) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.h]: Done

File: inference-engine/tests/functional/shared_test_classes/src/subgraph/simple_if.cpp [hunk 4]
```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/subgraph/simple_if.hpp"
+#include "ngraph_functions/builders.hpp"
+
+namespace SubgraphTestsDefinitions {
+std::string SimpleIfTest::getTestCaseName(const testing::TestParamInfo<SimpleIfParamsTuple> &obj) {
```
[the remark this answers is not in the document]
Comment (nshchego) [reply inference-engine/tests/functional/shared_test_classes/src/subgraph/simple_if.cpp:9]: Added

**Comment by dmitry-gorokhov** [comment]
@mandrono Do you have any additional comments?

**Review by mandrono — APPROVED** [review]

**Comment by evolosen** [comment]
@GlebKazantaev, @itikhono Please review ngraph part .

**Pull request merged**